plugins/pulp_rpm/plugins/importers/yum/existing.py

```diff
@@ -118,11 +118,17 @@ def check_all_and_associate(wanted, conduit, download_deferred, catalog):
         # fields = model.unit_key_fields + ('_storage_path',)
         unit_generator = (model(**unit_tuple._asdict()) for unit_tuple in values.copy())
         for unit in units_controller.find_units(unit_generator):
-            # Existing RPMs, DRPMs and SRPMs are disqualified when the associated
-            # package file does not exist and downloading is not deferred.
-            if not download_deferred and unit_type in (
-                    ids.TYPE_ID_RPM, ids.TYPE_ID_SRPM, ids.TYPE_ID_DRPM):
-                if unit._storage_path is None or not os.path.isfile(unit._storage_path):
+            package_type = unit_type in (ids.TYPE_ID_RPM, ids.TYPE_ID_SRPM, ids.TYPE_ID_DRPM)
```

Comment:
I like this re-factor to simplify the logic by breaking it up into smaller expressions. But I'll suggest a change of name. ``package_type`` sounds from its name like its value would be some identifier for a type. But instead, it is a bool. Perhaps a name like this would be more natural and meaningful: ``is_rpm_srpm_drpm``

Prefixing with ``is_`` is a nice hint that the value is a bool.